### ISSUE TITLE
Several gameplay problems

### ISSUE THREAD

**Issue by ObaniGemini** [MinetestForFun/server-minetestforfun-magichet#24]
By playing I discovered several gameplay problems : 
- Normal trees are very difficult to find (this is not so much a problem, but the next is)
- Normal trees don't drop saplings (cut three ones and no saplings...)
- Cobble isn't dropped everytime ?
- TOO MANY MONSTERS : This even worst than when we were talking about it on mff official server.
- Monsters AI : Monsters have a very strange way to work.
- I still can't understand how to recover ghost blocks, has if I already had some.
- Doc isn't really clear. Also, shouldn't we have a way to craft or to get the guide book ? And shouldn't we remove this screen to choose between Russian and English ?
- Too many empty biomes, it's very difficult to get ressources and very long to travel.
- Sometimes oceans aren't generated properly, and are brutally cut (making some 8/10 blocks falls in a desert land)
- Some blocks, when destroyed, become a 10 HP entity, destroyable with right click (one shot) or killable.

Please tell me what you think of it, and ask me if you want screenshots

**Comment by Ombridride** [comment]
This issue isn't good. you are talking about too many point, you need to create one issue per subject (biome / monsters / etc)

**Comment by Ombridride** [comment]
Assignee @4aiman because it's an Upstream issue and he is the best dev to fix it because he knows the mods better than us

**Comment by 4aiman** [comment]
Saplings are to be dropped from the leaves.
Dig with a right tool and you'll get your drops.
Magichet does drops MC-way: dig anything with everything , but get drops only if you're doing it right.
The digging times are from MC too (~97% of what is present in both games).

**Comment by 4aiman** [comment]
Oh...
This IS fixed in dev..

**Comment by 4aiman** [comment]
diff --git a/mods/default/leaves.lua b/mods/default/leaves.lua
index 62d9c56..ab7c5ea 100644
@@ -35,17 +35,17 @@ minetest.register_node(":"..name, {
         local dropped = false
         local wielded = digger:get_wielded_item():get_name()
         if cutby then
-        --minetest.chat_send_all(wielded)
-        for i,tool in ipairs(cutby) do
-            --minetest.chat_send_all(wielded .. '; '..tool..'; '..minetest.serialize(minetest.registered_items[wielded][tool]))
-            if wielded == tool
-            or minetest.get_item_group(wielded, tool)>0
-            then
-               dropped = true
-               --minetest.chat_send_all('wielded: '.. wielded ..', tool/group: '..tool)
-               break
-            end
-        end
+			--minetest.chat_send_all(wielded)
+			for i,tool in ipairs(cutby) do
+				--minetest.chat_send_all(wielded .. '; '..tool..'; '..minetest.serialize(minetest.registered_items[wielded][tool]))
+				if wielded == tool
+				or minetest.get_item_group(wielded, tool)>0
+				then
+				   dropped = true
+				   --minetest.chat_send_all('wielded: '.. wielded ..', tool/group: '..tool)
+				   break
+				end
+			end
         else
             if name:find('mushroom') then dropped=true end
         end
@@ -80,57 +80,62 @@ end
 register_leaves("default:leaves",
                 "Leaves",
                 {"default_leaves.png"},
-                {{name='ethreal:tree_sapling',chance=20},{name='default:apple',chance=100}},
-                {"default:shears"})
+                {{name='default:tree_sapling',chance=20},{name='default:apple',chance=50}},
+                {"4tools:shears"},
+                {cutty=3000, leaves=1})
 -- dug by shears
 register_leaves("default:jungleleaves",
                 "Jungle Tree Leaves",
                 {"default_leaves.png"},
                 {{name='default:jungle_tree_sapling',chance=20}},
-                {"default:shears"})
+                {"4tools:shears"},
+                {cutty=3000, leaves=1})
 -- dug by sharp and shears
 register_leaves("default:bananaleaves",
                 "Banana Leaves",
                 {"banana_leaf.png"},
                 {{name='default:banana_tree_sapling',chance=20}},
-                {"default:shears", "sharp"})
+                {"4tools:shears", "sharp"},
+                {cutty=3000, leaves=1})
 -- dug by sharp, cutters and shears
 register_leaves("default:yellowleaves",
                 "Yellow Leaves",
-                {"default_leaves.png"},
+                {"default_yellowleaves.png"},
                 {{name='default:apple',chance=30},{name='default:yellow_tree_sapling',chance=10}},
-                {"default:shears", "cutters"},
-                {snappy=3000, leaves=1})
+                {"4tools:shears", "cutters", "sharp"},
+                {cutty=13000, leaves=1})
 -- dug by shears
 register_leaves("default:palmleaves",
                 "Palm Tree Leaves",
                 {"moretrees_palm_leaves.png"},
                 {{name='default:palm_sapling',chance=20}},
-                {"default:shears"})
+                {"4tools:shears"},
+                {cutty=3000, leaves=1})
 -- dug by shears and cutters
 register_leaves("default:pineleaves",
                 "Pine Needles",
                 {"pine_leaves.png"},
                 {{name='default:pine_tree_sapling',chance=20},{name='default:pine_nuts',chance=5}},
-                {"default:shears", "cutters"})
+                {"4tools:shears", "cutters"},
+                {cutty=3000, leaves=1})
 -- dug by cutters
 register_leaves("default:frost_leaves",
                 "Frost Leaves",
                 {"default_frost_leaves.png"},
                 {{name='default:frost_tree_sapling',chance=20}},
                 {"cutters"},
-                {cracky=10000, puts_out_fire=1})
+                {choppy=10000, puts_out_fire=1})
 -- dug by ...
 register_leaves("default:mushroom",
                 "Mushroom Cap",
                 {"mushroom_block.png"},
                 {{name='default:mushroom_sapling',chance=20}},
-                nil,
-                {snappy=2000, oddly_breakable_by_hand=1})
+                {"4tools:shears"},
+                {cutty=2000})
 -- dug by ...
 register_leaves("default:mushroom_pore",
                 "Mushroom Pore",
                 {"mushroom_block2.png"},
                 {},
-                nil,
-                {snappy=2000, leaves=1, bouncy = 50, slippery = 70})
+                {"4tools:shears"},
+                {cutty=2000, leaves=1, bouncy = 50, slippery = 70})

**Comment by Ombridride** [comment]
As we will only takes your code from "magichet-today" version, can you add this code to the master "magichet-today" branch please ?

In my memory, I don't thing i taken your subgame from the "magichet-stable" repository, so i don't understand why we don't have this fix...

**Comment by 4aiman** [comment]
You're right you didn't fork MGC repo, so I see no use in adding the fix in question to master.
Besides, there are only 3 people (including me) with the acces to the latest dev and I'm not the one who is going to increase the number of devs.

I can attach the latest version of a file here, but I do not guarantee it's compatible.
Will that suffice?
